### PR TITLE
Refactor/loggedin folder

### PR DESCRIPTION
**Pull request by SteRiccio**
Most of the changes are related to the moving of files from webapp/loggedin into webapp/components
Only this file has been completely refactored: webapp/components/survey/Record/RecordView.js

### CODE REVIEW

File: webapp/components/survey/Record/RecordView.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+
+import SurveyFormView from '@webapp/components/survey/SurveyForm'
+import { useRecordView } from './useRecordView'
+
+const RecordView = () => {
```
Comment (minotogna) [inline webapp/components/survey/Record/RecordView.js:6]: rename this to Record

File: webapp/components/survey/Record/RecordView.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  const { canEditRecord, preview, recordLoaded } = useRecordView()
+
```
Comment (minotogna) [inline webapp/components/survey/Record/RecordView.js:8]: better to have ```if(!recordLoaded) return null```

File: webapp/components/survey/Record/useRecordView.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+import * as AppWebSocket from '@webapp/app/appWebSocket'
+import { RecordActions, RecordState } from '@webapp/store/ui/record'
+import { useSurveyInfo, useSurveyCycleKey } from '@webapp/store/survey'
+import { SurveyFormActions } from '@webapp/store/ui/surveyForm'
+import { useAuthCanEditRecord } from '@webapp/store/user'
+
+import { useOnUpdate } from '@webapp/components/hooks'
+
+export const useRecordView = () => {
```
Comment (minotogna) [inline webapp/components/survey/Record/useRecordView.js:18]: use local state convention

File: webapp/store/ui/record/actions.js [hunk 4]
```diff
@@ -9,15 +9,14 @@ import * as NodeRefData from '@core/record/nodeRefData'
 
 import { debounceAction } from '@webapp/utils/reduxUtils'
 
+import { AppSavingActions } from '@webapp/store/app'
 import { SurveyState } from '@webapp/store/survey'
-
-
+import { UserState } from '@webapp/store/user'
 import { LoaderActions, NotificationActions } from '@webapp/store/ui'
-import { AppSavingActions } from '@webapp/store/app'
```
Comment (minotogna) [inline webapp/store/ui/record/actions.js:16]: would it be relatively quick to split Actions into separate files (create, read, update, delete) ?

File: webapp/views/App/views/Data/Data.js [hunk 5]
```diff
@@ -8,7 +8,7 @@ import { useSurveyInfo } from '@webapp/store/survey'
 
 import ModuleSwitch from '@webapp/components/moduleSwitch'
 import SurveyDefsLoader from '@webapp/components/survey/SurveyDefsLoader'
-import RecordView from '@webapp/loggedin/surveyViews/record/recordView'
+import RecordView from '@webapp/components/survey/Record'
```
Comment (minotogna) [inline webapp/views/App/views/Data/Data.js:11]: call this ```import Record ... ```

File: webapp/components/survey/Record/RecordView.js (hunk 1 above)
Comment (SteRiccio) [reply webapp/components/survey/Record/RecordView.js:6]: ok

File: webapp/components/survey/Record/useRecordView.js (hunk 3 above)
Comment (SteRiccio) [reply webapp/components/survey/Record/useRecordView.js:18]: ok

File: webapp/views/App/views/Data/Data.js (hunk 5 above)
Comment (SteRiccio) [reply webapp/views/App/views/Data/Data.js:11]: sure

File: webapp/components/survey/Record/RecordView.js (hunk 2 above)
Comment (SteRiccio) [reply webapp/components/survey/Record/RecordView.js:8]: ok

File: webapp/store/ui/record/actions.js (hunk 4 above)
Comment (SteRiccio) [reply webapp/store/ui/record/actions.js:16]: right

File: webapp/views/App/views/Data/Data.js (hunk 5 above)
Comment (SteRiccio) [reply webapp/views/App/views/Data/Data.js]: right

File: webapp/components/survey/Record/Record.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+
+import SurveyFormView from '@webapp/components/survey/SurveyForm'
```
Comment (minotogna) [inline webapp/components/survey/Record/Record.js:3]: I would call SurveyForm as well.. sorry missed that before

File: webapp/store/ui/record/actions/actionTypes.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+export const recordCreate = 'survey/record/create'
+export const recordLoad = 'survey/record/load'
+export const recordDelete = 'survey/record/delete'
+export const recordUuidPreviewUpdate = 'survey/record/preview/update'
+export const nodesUpdate = 'survey/record/node/update'
+export const nodeDelete = 'survey/record/node/delete'
+export const validationsUpdate = 'survey/record/validation/update'
```
Comment (minotogna) [inline webapp/store/ui/record/actions/actionTypes.js:7]: I would remove survey prefix. I'd rather add ui.

Comment (SteRiccio) [reply webapp/store/ui/record/actions/actionTypes.js:7]: right!

File: webapp/components/survey/Record/Record.js (hunk 6 above)
Comment (SteRiccio) [reply webapp/components/survey/Record/Record.js:3]: sure, my mistake

**Pull request merged**